### PR TITLE
Implement throwing const re-assignment error.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-get-put-value.c [hunk 1]
```diff
@@ -257,9 +257,19 @@ ecma_op_put_value_lex_env_base (ecma_object_t *lex_env_p, /**< lexical environme
 
             ecma_named_data_property_assign_value (lex_env_p, property_value_p, value);
           }
-          else if (is_strict)
+          else
```
Comment (galpeter) [inline jerry-core/ecma/operations/ecma-get-put-value.c:260]: Do we need the extra indentation at all? Would it be possible to do something like this?
```c
#if ENABLED (JERRY_ES2015)
    else if (ecma_is_property_enumerable (*property_p))
    {
      // ...
    }
#endif /* ENABLED (JERRY_ES2015) */
    else if (is_strict)
    {
      // ....
    }
```

File: jerry-core/parser/js/js-parser.c [hunk 2]
```diff
@@ -2468,11 +2468,9 @@ parser_parse_arrow_function (parser_context_t *context_p, /**< context */
   JERRY_ASSERT (status_flags & PARSER_IS_ARROW_FUNCTION);
   parser_save_context (context_p, &saved_context);
   context_p->status_flags |= status_flags;
-#if ENABLED (JERRY_ES2015)
```
[the remark this answers is not in the document]
Comment (zherczeg) [reply jerry-core/parser/js/js-parser.c:2471]: I think they are simple enough.

File: jerry-core/parser/js/js-parser-expr.c [hunk 3]
```diff
@@ -2437,20 +2469,40 @@ parser_process_binary_opcodes (parser_context_t *context_p, /**< context */
...
+      if (index >= 0)
       {
-        uint16_t index = parser_stack_pop_uint16 (context_p);
-        parser_emit_cbc_literal (context_p, (uint16_t) opcode, index);
+        if (context_p->last_cbc_opcode == CBC_PUSH_LITERAL
+            && opcode == CBC_ASSIGN_SET_IDENT)
+        {
+          JERRY_ASSERT (CBC_ARGS_EQ (CBC_ASSIGN_LITERAL_SET_IDENT,
+                                     CBC_HAS_LITERAL_ARG | CBC_HAS_LITERAL_ARG2));
+          JERRY_ASSERT (index >= 0);
```
[the remark this answers is not in the document]
Comment (zherczeg) [reply jerry-core/parser/js/js-parser-expr.c:2498]: Good catch. The `index` is the result of `parser_stack_pop_uint16`, so it must be a 16 bit value.

**Comment by zherczeg** [comment]
Thank you for the review, patch is updated.

**Pull request merged**